### PR TITLE
InfluxDB: Fix for wrong query generated with template variable and non regex operator on frontend mode

### LINKED ISSUE #81517: InfluxDB: Wrong query generated when using templateVariable with equal operator on influxql visual query builder
When you are using a single value regex variable with a non regex operator (e.g. =), and you convert the query from builder to code-editor, it leaves the regex wrapper around the variable in the expression, and wraps that in a single quote, which leads to a broken query expression:

Image

Image

### PR DESCRIPTION
**Pull request by wasim-nihal**
**What is this feature?**

When using a single value regex variable with a non-regex operator (e.g. `=`), and you convert the query from builder to code editor, it leaves the regex wrapper around the variable in the expression, and wraps that in a single quote, which leads to a broken query expression. This PR is fixing it. 

**Why do we need this feature?**

Better query generation while switching between editors

**Who is this feature for?**

InfluxDB users building InfluxQL queries on grafana

**Which issue(s) does this PR fix?**: 

Fixes https://github.com/grafana/grafana/issues/81517

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by itsmylife** [comment]
@wasim-nihal The PR looks good. But in the CI there are some errors. Could you please run `yarn prettier:write` and commit the changes?

**Comment by wasim-nihal** [comment]
> @wasim-nihal The PR looks good. But in the CI there are some errors. Could you please run `yarn prettier:write` and commit the changes?

sure.

**Comment by wasim-nihal** [comment]
Done. Could you please re-trigger?

**Review by NWRichmond — APPROVED** [review]
Nice! It appears to work as expected:

https://github.com/grafana/grafana/assets/5732000/3882324f-f46b-4c75-ba51-3e8b07ce9956

I made a few suggestions. Probably the most pressing one is fixing the ~`infuxdb`~ > `InfluxDB` spelling issues.

**Review by itsmylife — APPROVED** [review]
LGTM 

Could you please check @NWRichmond 's comments too? Then I think we are good to go. Thanks for this contribution!

File: public/app/plugins/datasource/influxdb/influx_query_model.test.ts [hunk 1]
```diff
@@ -604,4 +604,141 @@ describe('InfluxQuery', () => {
       });
     });
   });
+
+  describe('test query generated with templateVariable and non-regex operator', () => {
```
[the remark this answers is not in the document]
Comment (wasim-nihal) [reply public/app/plugins/datasource/influxdb/influx_query_model.test.ts:608]: Incorporated this suggestion in the latest commit.

**Comment by wasim-nihal** [comment]
@itsmylife , @NWRichmond , thanks for reviewing this change. I have addressed the comments in the latest commit.

**Comment by wasim-nihal** [comment]
Seeing some linting issues again in CI. Will push the changes.

**Comment by wasim-nihal** [comment]
> Seeing some linting issues again in CI. Will push the changes.

Done

**Comment by itsmylife** [comment]
I triggered the build and waiting for it to start. Thanks for the fixes.

**Comment by itsmylife** [comment]
@wasim-nihal Thank you for the contribution once more! 🚀

**Pull request merged**